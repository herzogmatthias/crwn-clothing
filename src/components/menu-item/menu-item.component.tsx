import * as React from "react";
import { RouteComponentProps, withRouter } from "react-router-dom";
import {
  MenuItemContainer,
  BackgroundImageContainer,
  ContentContainer,
  TitleContainer,
  SubtitleContainer,
} from "./menu-item.styles";

export interface IMenuItemProps extends RouteComponentProps<{}> {
  title: string;
  imageUrl: string;
  linkUrl: string;
  size?: string;
}

export function MenuItem({
  title,
  imageUrl,
  size,
  history,
  linkUrl,
  match,
}: IMenuItemProps) {
  return (
    <MenuItemContainer
      size={size}
      onClick={() => history.push(`${match.url}${linkUrl}`)}
    >
      <BackgroundImageContainer url={imageUrl}></BackgroundImageContainer>
      <ContentContainer className="content">
        <TitleContainer className="title">{title.toUpperCase()}</TitleContainer>
        <SubtitleContainer className="subtitle">SHOP NOW</SubtitleContainer>
      </ContentContainer>
    </MenuItemContainer>
  );
}

export default withRouter(MenuItem);
