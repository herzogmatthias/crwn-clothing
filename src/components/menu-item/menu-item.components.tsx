import * as React from "react";
import "./menu-item.styles.scss";

export interface IMenuItemProps {
  title: string;
  imageUrl: string;
  size?: string;
}

export default function MenuItem({ title, imageUrl, size }: IMenuItemProps) {
  return (
    <div
      style={{
        backgroundImage: `url(${imageUrl})`,
      }}
      className={`${size} menu-item`}
    >
      <div className="content">
        <h1 className="title">{title}</h1>
        <span className="subtitle">SHOP NOW</span>
      </div>
    </div>
  );
}
