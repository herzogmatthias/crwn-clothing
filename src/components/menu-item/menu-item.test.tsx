import React from "react";
import { shallow, ShallowWrapper } from "enzyme";

import { MenuItem } from "./menu-item.component";
import {
  MenuItemContainer,
  BackgroundImageContainer,
} from "./menu-item.styles";

describe("NOT CONNECTED MENU-ITEM COMPONENT", () => {
  let wrapper: ShallowWrapper;
  let mockMatch: any;
  let mockHistory;
  const linkUrl = "/hats";
  const size = "large";
  const imageUrl = "testimage";

  beforeEach(() => {
    mockMatch = {
      url: "/shop",
    };

    mockHistory = {
      push: jest.fn(),
    } as any;

    const mockLocation = {} as any;
    const mockProps = {
      match: mockMatch,
      history: mockHistory,
      location: mockLocation,
      linkUrl,
      size,
      title: "hats",
      imageUrl,
    };

    wrapper = shallow(<MenuItem {...mockProps} />);
  });

  test("render MenuItem component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("call history.push with the right string when MenuItemContainer clicked", () => {
    wrapper.find(MenuItemContainer).simulate("click");

    expect(mockHistory.push).toHaveBeenCalledWith(`${mockMatch.url}${linkUrl}`);
  });

  test("pass size to MenuItemContainer as the prop size", () => {
    expect(wrapper.find(MenuItemContainer).prop("size")).toBe(size);
  });

  test("pass imageUrl to BackgroundImageContainer as the prop imageUrl", () => {
    expect(wrapper.find(BackgroundImageContainer).prop("url")).toBe(imageUrl);
  });
});
