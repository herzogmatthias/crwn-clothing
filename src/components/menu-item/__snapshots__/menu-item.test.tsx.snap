// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem component render MenuItem component 1`] = `
<styled.div
  onClick={[Function]}
  size="large"
>
  <styled.div
    url="testimage"
  />
  <styled.div
    className="content"
  >
    <styled.h1
      className="title"
    >
      HATS
    </styled.h1>
    <styled.span
      className="subtitle"
    >
      SHOP NOW
    </styled.span>
  </styled.div>
</styled.div>
`;
